File: paiza-js/src/mondai/B_rank/class_prime/class_primer__change/class_primer__change.js
import fs from 'fs';
import splitTokens from '../../../../util/splitTokens-js/src/splitTokens.js'

// 入力処理
const input = splitTokens(fs.readFileSync(0, 'utf8')); //const input = splitTokens(require('fs').readFileSync(0, 'utf8'));
const [[n], ...rest] = input;

// データ変換処理（配列→構造体）
const members = rest.slice(0, n).map(([nickname, old, birth, state]) => (
  { nickname, old, birth, state }
));

// データ更新処理（nicknameを更新）
rest.slice(n).forEach(([i, name]) => {
  members[i - 1].nickname = name
});

// 出力処理
members.forEach(({ nickname, old, birth, state }) => {
  console.log(`${nickname} ${old} ${birth} ${state}`)
});

File: paiza-py/src/mondai/B_rank/class_prime/class_primer__change/class_primer__change.py
import sys, os

sys.path.append(os.path.join(os.path.dirname(__file__), "../../../.."))  # src を追加
from util.splitTokens_py.src.splitTokens import splitTokens

# 入力処理
[n, _], *rest = splitTokens(sys.stdin.read())

# データ変換処理（配列→構造体（辞書型））
members = [dict(zip(["nickname", "old", "birth", "state"], row)) for row in rest[:n]]

# データ更新処理（nicknameを更新）
for a, nn in rest[n:]:
    members[a - 1]["nickname"] = nn

# 出力処理
for m in members:
    print(f'{m["nickname"]} {m["old"]} {m["birth"]} {m["state"]}')
